### ISSUE TITLE
Appium iOS crash - Instrument Server Socket

### ISSUE THREAD

**Issue by brunosmm** [appium/appium#6690]
## The problem

Once you ask to execute the second session on the same port the appium crashes:
2016-07-22 18:26:42:022 - [debug] [UIAuto] Instruments socket server started at /var/folders/6r/dv8krnss46lcq0y_1cbvyhd00000gn/T/temp_86f5533b9eb2358fe310039d57533c4138de20bf4222923236634886567/instruments_sock
Error: listen EADDRINUSE /var/folders/6r/dv8krnss46lcq0y_1cbvyhd00000gn/T/temp_86f5533b9eb2358fe310039d57533c4138de20bf4222923236634886567/instruments_sock
    at Object.exports._errnoException (util.js:890:11)
    at exports._exceptionWithHostPort (util.js:913:20)
    at Server.__dirname.Server.Server._listen2 (net.js:1217:19)
    at listen (net.js:1266:10)
    at Server.__dirname.Server.Server.listen (net.js:1356:5)
    at UIAutoClient.start$ (lib/uiauto-client.js:145:23)
    at tryCatch (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke](/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next](/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at run (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:104:47)
    at /Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:115:28
    at flush (/Applications/Appium.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/$.microtask.js:19:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
## Environment
- Appium version (or git revision) that exhibits the issue: 1.5.3
- Desktop OS/version used to run Appium: OS X 10.11.4
- Node.js version (unless using Appium.app|exe): using appium.app
- Mobile platform/version under test: iOS 
- Real device or emulator/simulator: real device
- Appium CLI or Appium.app|exe: Appium.app
## Details

Create appium 2 sessions via nodejs, using the parameters:
-bp, -p, -tmp
The idea is execute two real iOS devices in parallel(https://github.com/appium/appium/issues/5925)
## Link to Appium logs

Logs from first execution(Works normally), and second execution straight after the other.
https://gist.github.com/brunosmm/f628d933f450be414a237a5c2eb8f933

**Comment by imurchie** [comment]
Full support for parallel tests in one server is not available at this time. To do this you need to run different Appium servers.

**Comment by brunosmm** [comment]
Hi @imurchie ,

I am running in two different server, I start two appium servers.
The problem I have is once the server finishes and I start a new session on the same server the instruments crash.

**Comment by imurchie** [comment]
Ah.

Interesting. Can you post the logs from the concurrent session in the other instance of Appium? It seems that there is a shared resource that should not be shared.

**Comment by brunosmm** [comment]
Actually that is not even the case(I think), as I am using just one server to execute in sequencial the other server has no logs.
The logs attach is from the current scenario:

1. Start two appium servers;
2. Execute a test in appium server 1, server 2 is  not being used;
3. Wait the execution finished and the session ended;
4. Execute the same test in appium server 1;
5. Instruments Server Socket crash;

**Comment by imurchie** [comment]
Does the problem persist if the second server is not around?

**Comment by brunosmm** [comment]
Yes.
The problems was introduce once I start using the --tmp to start appium servers.

**Comment by imurchie** [comment]
Ok. I'll look into this problem. Thanks!

**Comment by brunosmm** [comment]
Some udpate, I create some Java code to start the appium servers.
In this java code I created a temp dir and use this path to --tmp in appium, once I remove that part and change to "/tmp/appium/$port" appium handle to create the folder and the problem did not happen again.
I believe that was a problem to use the directory I created, but the funny thing was it work for the first execution but not for the second.
Any ideas?
